espnet2/tasks/abs_task.py

```diff
@@ -228,20 +257,55 @@ def get_parser(cls) -> configargparse.ArgumentParser:
             help="The number images to plot the outputs from attention. "
             "This option makes sense only when attention-based model",
         )
+
+        group = parser.add_argument_group("distributed training related")
         group.add_argument(
-            "--train_dtype",
-            default="float32",
-            choices=["float16", "float32", "float64", "O0", "O1", "O2", "O3"],
-            help="Data type for training. O0,O1,.. flags require apex. "
-            "See https://nvidia.github.io/apex/amp.html#opt-levels",
+            "--distributed",
+            type=str2bool,
+            default=False,
+            help="Enable distributed training",
         )
         group.add_argument(
-            "--log_interval",
-            type=int_or_none,
-            default=None,
-            help="Show the logs every the number iterations in each epochs at the "
-            "training phase. If None is given, it is decided according the number "
-            "of training samples automatically .",
+            "--dist_backend", default="nccl", type=str, help="distributed backend",
+        )
+        group.add_argument(
+            "--dist_init_method",
+            type=str,
+            default="env://",
+            help='if init_method="env://", env values of "MASTER_PORT", "MASTER_ADDR", '
+            '"WORLD_SIZE", and "RANK" are referred.',
```

Comment:
By default, rank is passed as —local_rank.
(This is just a note)
https://github.com/pytorch/pytorch/blob/34561dadcddf6ce3c76daf16f09a09adc9c7b73b/torch/distributed/launch.py#L179